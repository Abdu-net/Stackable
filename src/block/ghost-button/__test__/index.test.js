import { edit, save } from '../index'

describe( 'Ghost Button', () => {
	test( 'block edit matches snapshot', () => {
		const wrapper = edit( {
			attributes: {
				url: 'http://www.gambit.ph/',
				text: 'Button',
				textAlignment: 'center',
				color: '#ffffff',
				textColor: '#2091e1',
				size: 'normal',
				cornerButtonRadius: '4',
				borderThickness: '1',
			},
		} )

		expect( wrapper ).toMatchSnapshot()
	} )

	test( 'block save matches snapshot', () => {
		const wrapper = save( {
			attributes: {
				url: 'http://www.gambit.ph/',
				text: 'Button',
				textAlignment: 'center',
				color: '#ffffff',
				textColor: '#2091e1',
				size: 'normal',
				cornerButtonRadius: '4',
				borderThickness: '1',
			},
		} )

		expect( wrapper ).toMatchSnapshot()
	} )
} )
