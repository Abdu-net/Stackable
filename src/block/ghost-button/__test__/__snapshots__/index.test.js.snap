// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ghost Button block edit matches snapshot 1`] = `
<Fragment>
  <BlockControls>
    <BlockAlignmentToolbar
      controls={
        Array [
          "left",
          "center",
          "right",
          "full",
        ]
      }
      onChange={[Function]}
      value={undefined}
    />
  </BlockControls>
  <InspectorControls>
    <PanelBody>
      <SelectControl
        label="Size"
        onChange={[Function]}
        options={
          Array [
            Object {
              "label": "Small",
              "value": "small",
            },
            Object {
              "label": "Normal ",
              "value": "normal",
            },
            Object {
              "label": "Medium",
              "value": "medium",
            },
            Object {
              "label": "Large",
              "value": "large",
            },
          ]
        }
        value="normal"
      />
      <RangeControl
        label="Corner Radius"
        max="50"
        min="1"
        onChange={[Function]}
        value="4"
      />
      <RangeControl
        label="Border Thickness"
        max="10"
        min="1"
        onChange={[Function]}
        value="1"
      />
    </PanelBody>
    <PanelColorSettings
      colorSettings={
        Array [
          Object {
            "label": "Button Color",
            "onChange": [Function],
            "value": "#ffffff",
          },
        ]
      }
      title="Color Settings"
    />
  </InspectorControls>
  <span
    className="ugb-button ugb-ghost-button ugb-button-undefined ugb-button-normal"
    style={
      Object {
        "borderColor": "#ffffff",
        "borderRadius": "4px",
        "borderWidth": "1px",
      }
    }
  >
    <RichText
      className="ugb-button-inner"
      formattingControls={
        Array [
          "bold",
          "italic",
          "strikethrough",
        ]
      }
      keepPlaceholderOnFocus={true}
      onChange={[Function]}
      placeholder="Enter Text"
      style={
        Object {
          "color": "#ffffff",
        }
      }
      tagName="span"
      value="Button"
    />
  </span>
</Fragment>
`;

exports[`Ghost Button block save matches snapshot 1`] = `
<div
  className="ugb-button ugb-ghost-button ugb-button-undefined ugb-button-normal"
  style={
    Object {
      "borderColor": "#ffffff",
      "borderRadius": "4px",
      "borderWidth": "1px",
    }
  }
>
  <Unknown
    className="ugb-button-inner"
    href="http://www.gambit.ph/"
    style={
      Object {
        "color": "#ffffff",
      }
    }
    tagName="a"
    value="Button"
  />
</div>
`;
